lib/matplotlib/path.py

```diff
@@ -173,9 +178,15 @@ def _fast_from_codes_and_verts(cls, verts, codes, internals_from=None):
             ``should_simplify``, ``simplify_threshold``, and
             ``interpolation_steps`` will be copied.  Note that ``readonly`` is
             never copied, and always set to ``False`` by this constructor.
+        unmask_verts : bool
+            If False, vertices should be an unmasked NumPy array with dtype
+            set to Path.verts_type.
         """
         pth = cls.__new__(cls)
-        pth._vertices = _to_unmasked_float_array(verts)
+        if unmask_verts:
+            pth._vertices = _to_unmasked_float_array(verts)
```

Comment:
Apparently this function is not as fast due to an extra call overhead, and because `_to_unmasked_float_array` calls `np.asarray`, which is pretty expensive even when you pass an array in!

Reply:
See https://github.com/numpy/numpy/issues/13580 :) how much do we get if we just add in `_to_unmasked_float_array` a check to return things as if the arg is already an array?
Also, I think the warrants a comment explaining that this is for perf. reasons.

Reply:
I'm pretty sure that the extra call adds a fair amount of overhead itself, and a robust check would not be cheap either. I can try and circle back with the results if you want thought.